### PR TITLE
Reuse appgate client

### PR DESCRIPTION
**Pull request by lepiolet**
We were creating a new one every time we were entering the loop

This PR tries to reuse it as much as possible (it implements support for renewal of tokens)

### CODE REVIEW

**Review by thomascellerier — APPROVED** [review]
Additionally `self.request` could handle a 401 error and try re-authenticating, in case the timing is not accurate.

**Review by mandopaloooza — APPROVED** [review]

**Review by mandopaloooza — APPROVED** [review]

File: appgate/__main__.py [hunk 1]
```diff
@@ -245,36 +246,56 @@ def main_run(args: OperatorArguments) -> None:
 async def dump_entities(
     ctx: Context, output_dir: Optional[Path], stdout: bool = False
 ) -> None:
-    current_appgate_state = await get_current_appgate_state(ctx)
-    expected_appgate_state = AppgateState(
-        {
-            k: v.entities_with_tags(
-                ctx.builtin_tags.union(ctx.exclude_tags or frozenset())
+    async with AsyncExitStack() as exit_stack:
```
Comment (thomascellerier) [inline appgate/__main__.py:249]:
```suggestion
    async with AppgateClient() as appgate_client:
```

File: appgate/appgate.py [hunk 2]
```diff
@@ -406,56 +405,53 @@ async def main_loop(
...
                 log.info(
                     "[appgate-operator/%s] Nothing changed! Keeping watching!",
                     namespace,
                 )
+
+
+async def main_loop(
+    queue: Queue, ctx: Context, k8s_configmap_client: K8SConfigMapClient
+) -> None:
+    async with AsyncExitStack() as exit_stack:
```
Comment (thomascellerier) [inline appgate/appgate.py:440]:
```suggestion
    async with AppgateClient() as appgate_client:
```

**Review by thomascellerier — APPROVED** [review]
I don't think you need the exit stack now, just use the context manager directly.

**Pull request merged**